script.js

```diff
@@ -0,0 +1,146 @@
+'use strict';
+
+function main() {
+  const formBox = document.forms['reg-form'];
+  const extra = formBox.querySelector('#extra-info');
+  const regionSelect = formBox['region'];
+  const selectBox = formBox.querySelector('#select-box');
+  const privacy = formBox['privacy-check'];
+  const submit = formBox['submit'];
+  const regions = {
+    Kyiv: true,
+    Center: ['Cherkasy', 'Dnipro', 'Kropyvnytskyi', 'Poltava', 'Vinnytsia', 'Zhytomyr'],
+    North: ['Chernihiv', 'Sumy'],
+    East: ['Donetsk', 'Kharkiv', 'Luhansk'],
+    South: ['Kherson', 'Mykolaiv', 'Odesa', 'Zaporizhzhia'],
+    West: ['Chernivtsi', 'Ivano-Frankivsk', 'Khmelnytskyi', 'Lutsk', 'Lviv', 'Rivne', 'Ternopil', 'Uzhhorod']
+  };
+  const ignoreStrings = ['', 'Kyiv', 'Not in the list'];
+
+  submit.disabled = true;
+  formBox.addEventListener('input', checkInput);
+  addSelectList(Object.keys(regions), regionSelect);
+  regionSelect.addEventListener('change', selectHandle.bind(null, ignoreStrings, regions));
+  privacy.addEventListener('click', privacyHandle.bind(null, extra));
+  selectBox.addEventListener('change', checkInput);
+  formBox.addEventListener('change', validateForm.bind(null, submit));
+}
+
+const isValid = (stringToValidate, regex) => {
+  if (regex.test(stringToValidate)) {
+    return true;
+  }
+
+  return false;
+};
+
+const inputValidation = (input, normalizeString, type) => {
+  let regex;
+  switch(type) {
+    case 'full-name': 
+      regex = /^[a-zA-Zа-яА-ЯёЁїЇІі'\-]{2,30} [a-zA-Zа-яА-ЯёЁїЇІі'\-]{2,30} ?([a-zA-Zа-яА-ЯёЁїЇІі'\-]{2,30})$/;

```

Comment:
will be better to move regular expressions outside of the function and assign constants to regex variable here